pubtools/_pulp/tasks/delete/__init__.py

```diff
@@ -0,0 +1,15 @@
+from .base import multi_dispatch
+from .delete_packages import DeletePackages
+from .delete_advisory import DeleteAdvisory
+
+
+Delete = multi_dispatch("Delete", DeleteAdvisory, DeletePackages)
```

Comment:
I'm not sure if the multi_dispatch thing is the way to go here.  Could we just do all deletions that the user asked for?  i.e. why can't I delete an advisory and also delete some loose files at once?  It seems a bit of an artificial restriction that the code has to choose either to jump into a class which only supports deleting an advisory, or only supports deleting standalone files.

Reply:
I thought the same and then went ahead to split into two different classes as there's a difference in the approach of the two processes i.e. deleting standalone units and packages from advisory. However, the dispatch part is not working as expected as it looks like there's a deadlock as making the the dispatch decision and to instantiation of the class should be done on paring the args which again requires some instance of the class to utilize the existing parser in PulpTask and other parent classes.
So I believe going by the suggested approach to handle all the types in the same class makes sense. This will also give an option to do a mix-n-match deletion as mentioned above.
Taking this approach, can you please suggest if it should be an all-or-none removal or whatever applies for the repos to be removed from i.e. If a unit belong to two out of three provided repos, should the removal be skipped as the unit doesn't belong to all the repos or the unit be removed from the repos where it's present? Existing implementation in Pub has a mix of both.

Reply:
> If a unit belong to two out of three provided repos, should the removal be skipped as the unit doesn't belong to all the repos or the unit be removed from the repos where it's present?

I suggest the latter - it should be removed from the repos where it's present.  Otherwise the command wouldn't be retryable if interrupted partway through (e.g. imagine you ask it to remove a file from 3 repos, it succeeds from 2 and then crashes - you should be able to just retry your command without having to adjust the repo list to include just the final repo).

I think the command should be thought of more as "ensure content is absent", rather than "delete content".  So it's not an error if the content is already absent from some of the specified repos.  Like `rm -f somefile` does not fail if the file already is missing.